pandas/core/indexes/datetimes.py

```diff
@@ -806,8 +792,9 @@ def _mpl_repr(self):
 
     @cache_readonly
     def _is_dates_only(self):
+        """Return a boolean if we are only dates (and don't have a timezone)"""
         from pandas.io.formats.format import _is_dates_only
-        return _is_dates_only(self.values)
+        return _is_dates_only(self.values) and self.tz is None
```

Comment:
why did this need changing?

Reply:
After this refactor, I was getting errors where the repr didn't look correct but the i8 data was correct in the case when a tz was passed in the `DatetimeIndex` constructor.

`_is_dates_only` tries to:
> return a boolean if we are only dates (and don't have a timezone)

but checks for timezones by re-passing `self.values` into `DatetimeIndex`. I am not sure the historical reason why this was the case, but checking `self.tz is None` fixed the repr issue (and seems to achieve what that function was trying to do in the first place)